dart-if/include/dash/dart/if/dart_globmem.h

```diff
@@ -257,6 +257,89 @@ dart_ret_t dart_gptr_setflags(
   dart_gptr_t * gptr,
   uint16_t      flags) DART_NOTHROW;
 
+/**
+ * DART allocator used for non-collective global memory allocations using
+ * \ref dart_allocator_alloc similar to \ref dart_memalloc.
+ * An instance of the allocator has to be created using \c dart_allocator_new
+ * before allocating memory using \ref dart_allocator_alloc, which can then
+ * be shared with other units in the same team. The memory has to be free'd
+ * using \ref dart_allocator_free by the unit that allocated it.
+ * The allocator should be destroyed after all allocations have been free'd
+ * using \ref dart_allocator_destroy.
+ */
+typedef struct dart_allocator_struct * dart_allocator_t;
+
+/**
+ * Create a new allocator for non-collective global memory allocations.
+ * This operation is collective among the units in \c team.
+ *
+ * \param pool_size The size of the local memory pool from which global memory
+ *                  is allocated in \ref dart_allocator_alloc.
+ * \param team      The team describing the group of units.
+ * \param[out] new_allocator Pointer the newly created allocator.
+ *
+ * \return \c DART_OK on success, any other of \ref dart_ret_t otherwise.
+ *
+ * \threadsafe_data{team}
+ * \ingroup DartGlobMem
+ */
+dart_ret_t dart_allocator_new(
+  size_t             pool_size,
+  dart_team_t        team,
+  dart_allocator_t * new_allocator);
+
+/**
+ * Allocate global memory from a DART allocator instance created using
+ * \ref dart_allocator_new. The operation is not a collective operation but only
+ * involves the local unit. The resulting \ref dart_gptr_t can be communicated
+ * to other units for global memory access.
+ *
```

Comment:
Please add some notes on the alignment. Normally I would not bother, but we do HPC ;)

Reply:
We currently don't have any guarantees (unfortunately) on the memory allocation, mostly because MPI does not provide us with any guarantees and we haven't implemented it ourselves. I would thus not make any statements here (in reality, natural alignment should be guaranteed).